gadget/gadget.go

```diff
@@ -185,21 +185,35 @@ func (v *Volume) HasPartial(pp PartialProperty) bool {
 	return false
 }
 
-// MinSize returns the minimum size required by a volume, as implicitly
-// defined by the size structures. It assumes sorted structures.
-func (v *Volume) MinSize() quantity.Size {
+// size returns the size of the volume using the structureSizer function to calculate
+// structures size. It assumes sorted structures.
+func (v *Volume) size(structureSizer func(VolumeStructure) quantity.Size) quantity.Size {
 	endVol := quantity.Offset(0)
 	for _, s := range v.Structure {
 		if s.Offset != nil {
-			endVol = *s.Offset + quantity.Offset(s.MinSize)
+			endVol = *s.Offset + quantity.Offset(structureSizer(s))
 		} else {
-			endVol += quantity.Offset(s.MinSize)
+			endVol += quantity.Offset(structureSizer(s))
 		}
 	}
 
 	return quantity.Size(endVol)
 }
 
+// MinSize returns the minimum size required by a volume.
+func (v *Volume) MinSize() quantity.Size {
+	return v.size(func(s VolumeStructure) quantity.Size {
+		return s.MinSize
+	})
+}
+
+// MaxSize returns the maximum size required by a volume.
+func (v *Volume) MaxSize() quantity.Size {
```

Comment:
How about simply `Size()`, or `DefaultSize()` (to highlight that the volume can be expanded at install time)?

It's an odd choice for u-i to be using MinSize() really. AFAIU MinSize is there only for compatibility check during remodel/update.

Made a similar remark in the u-i PR https://github.com/canonical/ubuntu-image/pull/251

Reply:
It was more than an odd choice, it was a bug! 

I do not know if the gadget info is updated when the volume is expanded at install time, so I do not know if `DefaultSize()` would make sense or not (because if the gadget info is updated, then the value returned by `DefaultSize()` would not be the default anymore once the volume is expanded). But I think renaming this method to `Size()` makes sense. 

@alfonsosanchezbeato any opinion on that?

Reply:
Maybe `DefaultImageSize()` as it would be the default size of a flashable image?

Reply:
I renamed it `Size()` because this method could be used for other purposes than giving the "Image" size, and the returned value will change if the Volume was updated so I find it confusing to use `Default` in the name.